source/code-snippets/typescript/dot-notation.ts

```diff
@@ -15,11 +15,11 @@ const collection = database.collection<TestType>("<your collection>");
 await collection.updateOne({}, { $set: { field: { nested: "A string" } } });
 // end-error
 // start-no-key
-interface TestNumber {
-  myNumber: number;
+interface User {
+  email: string;
 }
 
-const database = client.db("<your db>");
-const collection = db.collection<TestNumber>("...");
-collection.find({ someRandomKey: "Accepts any type!" });
+const database = client.db("<your database>");
+const collection = db.collection<User>("<your collection>");
+collection.find({ age: "Accepts any type!" });
```

Comment:
Just a note that this will probably need to change soon to something like `{ $unknownOperator: "Accepts any type!" }` because we actually have a user submitted fix to make sure that we validate unknown keys that aren't $-prefixed but allow all $-prefixed keys for forward compatibility with new server operators. Right now the corresponding note is correct, but once we release a fix, it will need to be updated.

(see https://github.com/mongodb/node-mongodb-native/pull/3115)

Reply:
Hi Daria! There is a ticket to add the "[query nested fields without dot notation](https://jira.mongodb.org/browse/DOCSP-20479)" example back to a different section of the documentation (DevED-DBX agreed that TS page seemed only tangentially related to dot notation and was not the correct place for this content).

The page that should explain to readers how to use Dot Notation [is the Update Arrays in a Document Page](https://docs.mongodb.com/drivers/java/sync/current/fundamentals/crud/write-operations/embedded-arrays/#specifying-array-elements) (linking to Java), however it seems like the corresponding Node page does not include [explicitly discuss dot notation](https://docs.mongodb.com/drivers/node/current/fundamentals/crud/write-operations/embedded-arrays/). I've created a ticket[ to add this content ](https://jira.mongodb.org/browse/DOCSP-20598).

As this content has already been through a few internal reviews, I think it is best to close out this pull request and handle [the upcoming changes](https://github.com/mongodb/node-mongodb-native/pull/3115) as part of a new task (any substantial additions will require an additional copy review which will take a decent amount of time as this PR has grown fairly large). 

Sorry for the long response! Let me know if I have failed to address any of your concerns.

Reply:
Good point, I agree an example of working dot notation type checking would benefit readers. I've added a "[dot notation](https://docs-mongodbcom-staging.corp.mongodb.com/node/docsworker-xlarge/DOCSP-20071-Support-Dot-Notation-TS/fundamentals/typescript/#dot-notation)" subsection to the Features section Let me know if this addresses your concerns!